src/settings/settingsManager.ts

```diff
@@ -48,67 +26,31 @@ export type SettingsManagerView = Pick<
 
 let settingsManager: SettingsManagerView | undefined;
 
-export const DEFAULT_OPEN_AI_CHAT_MODEL = 'gpt-3.5-turbo';
-
 export async function initSettings() {
+  let settings: SettingsSchema;
   try {
-    const settings: SettingsSchema = {
-      project: {
-        currentDir: import.meta.env.VITE_IS_WEB ? '' : 'MIGRATE_FROM_GENERAL',
-      },
-      openAI: {
-        apiKey: await readEnv('OPENAI_API_KEY', ''),
-        chatModel: await readEnv('OPENAI_CHAT_MODEL', DEFAULT_OPEN_AI_CHAT_MODEL),
-        manner: (await readEnv('OPENAI_MANNER', 'scholarly')) as OpenAiManner,
-        temperature: parseFloat(await readEnv('OPENAI_TEMPERATURE', '0.7')),
-      },
-      sidecar: {
-        logging: {
-          active: (await readEnv('SIDECAR_ENABLE_LOGGING', 'false')).toLowerCase() === 'true',
-          path: await readEnv('SIDECAR_LOG_DIR', '/tmp'),
-        },
-      },
-    };
-
-    let configs;
-    if (import.meta.env.VITE_IS_WEB) {
-      configs = settings;
-
-      settingsManager = {
-        default: settings,
-        getCache: (key) => getDotNotation(settings, key)!,
-        setCache: (key, value) => {
-          setDotNotation(settings, key, value);
-          return value;
-        },
-        syncCache: () => Promise.resolve(settings),
-      };
-    } else {
-      const tauriSettingsManager = new SettingsManager<SettingsSchema>(settings, {
-        dir: await getSystemConfigurationsDir(),
-        fileName: 'refstudio-settings.json',
-        prettify: true,
-      });
-      settingsManager = tauriSettingsManager;
+    settings = (await universalGet('/api/settings/')) as SettingsSchema;
 
-      configs = await tauriSettingsManager.initialize();
-      // Run retro-compatibility migration if required key is missing
-      if (configs.project.currentDir === 'MIGRATE_FROM_GENERAL') {
-        if (configs.general?.appDataDir && configs.general.projectName) {
-          setCachedSetting('project.currentDir', configs.general.appDataDir + configs.general.projectName);
-        } else {
-          setCachedSetting('project.currentDir', '');
-        }
-        await saveCachedSettings();
-      }
-    }
-
-    console.log('Settings initialized with success with', configs);
-    console.log('openAI', configs.openAI);
+    console.log('Settings initialized with success with', settings);
+    console.log('openAI', settings.openai);
   } catch (err) {
     console.error('Cannot init settings', err);
     throw new Error('Cannot init settings');
   }
+
+  settingsManager = {
+    default: settings,
+    getCache: (key) => getDotNotation(settings, key)!,
+    setCache: (key, value) => {
+      setDotNotation(settings, key, value);
+      return value;
+    },
+    syncCache: async () => {
+      const newSettings = (await universalPost('/api/settings/', settings, 'PUT')) as SettingsSchema;
```

Comment:
Note: the trailing slash is important. If you leave it off, you get confusing CORS errors from the dev server in web mode.